core/src/main/java/io/grpc/inprocess/InProcessChannelBuilder.java

```diff
@@ -57,14 +57,22 @@
    * @return a new builder
    */
   public static InProcessChannelBuilder forName(String name) {
-    return new InProcessChannelBuilder(name);
+    return new InProcessChannelBuilder(new InProcessSocketAddress(name), null);
   }
 
-  private final String name;
+  static InProcessChannelBuilder forAnonymous(InProcessSocketAddress address,
+      InProcessInternalServer internalServer) {
```

Comment:
It seems `InProcessSocketAddress` ~ `InProcessInternalServer` ~ `InProcessServer` is one-to-one-to-one. Why do you need two arguments here? Couldn't just make `registry` a map of type `Map<InProcessSocketAddress, InProcessServer>`?

Reply:
I do not want anonymous servers to be in the registry.